backend/controllers/auth-controller.ts

```diff
@@ -38,20 +40,27 @@ export const signUpWithEmail = asyncHandler(async (req, res) => {
   try {
     await user.validate();
   } catch (error) {
-    const validationErrors = [];
-    for (const key in error.errors) {
-      validationErrors.push(error.errors[key].message);
+    if (error instanceof ApiError) {
+      const validationErrors = [];
+      for (const key in error.errors) {
+        validationErrors.push(error.errors[key].message);
+      }
+      throw new ApiError(HTTP_STATUS.BAD_REQUEST, validationErrors.join(', '));
+    } else {
+      throw new ApiError(
+        HTTP_STATUS.INTERNAL_SERVER_ERROR,
+        RESPONSE_MESSAGES.COMMON.INTERNAL_SERVER_ERROR
+      );
     }
-    throw new ApiError(HTTP_STATUS.BAD_REQUEST, validationErrors.join(', '));
   }
 
-  const accessToken = await user.generateAccessToken();
-  const refreshToken = await user.generateRefreshToken();
+  const accessToken: string = await user.generateAccessToken();
+  const refreshToken: string = await user.generateRefreshToken();
 
   user.refreshToken = refreshToken;
 
   await user.save();
-  user.password = undefined;
+  // user.password = undefined;
 
```

Comment:
Make the password field nullable in the type, so that you don't have to comment it for the ts error